src/main/java/org/takes/rq/RqLive.java

```diff
@@ -95,29 +97,16 @@ private static Request parse(final InputStream input) throws IOException {
                     break;
                 }
                 data = new Opt.Single<Integer>(input.read());
-                if (data.get() != ' ' && data.get() != '\t') {
-                    head.add(new String(baos.toByteArray()));
-                    baos.reset();
-                }
+                addHeader(data, baos, head);
```

Comment:
@HDouss it's a procedural design. Would be much better to have something like this here:

```
data = withHeader(data, baos, head);
```

See the difference? The same for your `legalCharacterCheck` below